# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContextFieldList.h
# Commit: 856a3d16a4ca
# Full Hash: 856a3d16a4ca856711ad9e6368cccd5aa1e22118
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-03-15 03:42:43
# Regressor Bug: 1532661
# File Overlap Count: 2
# Description:
#   Bug 1532661 - Part 5: Make the BrowsingContext opener edge a weak reference, r=farre
#   
#   Depends on D22193
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D23047
# ==============================================================================

diff -r 9ce1a2365aff -r 856a3d16a4ca docshell/base/BrowsingContextFieldList.h
--- a/docshell/base/BrowsingContextFieldList.h	Thu Mar 14 18:51:09 2019 +0000
+++ b/docshell/base/BrowsingContextFieldList.h	Thu Mar 14 18:51:11 2019 +0000
@@ -8,12 +8,9 @@
 MOZ_BC_FIELD(Closed, bool)
 MOZ_BC_FIELD(CrossOriginPolicy, nsILoadInfo::CrossOriginPolicy)
 
-// The current opener for this BrowsingContext. This field can be skipped by
-// defining `MOZ_BC_FIELD_SKIP_OPENER`, which is used when serializing
-// synchronized field state when initializing browsing contexts.
-#ifndef MOZ_BC_FIELD_SKIP_OPENER
-MOZ_BC_FIELD(Opener, RefPtr<BrowsingContext>)
-#endif
+// The current opener for this BrowsingContext. This is a weak reference, and
+// stored as the opener ID.
+MOZ_BC_FIELD(OpenerId, uint64_t)
 
 // Toplevel browsing contexts only. This field controls whether the browsing
 // context is currently considered to be activated by a gesture.